app/components/post_textbox/post_textbox_base.js

```diff
@@ -408,62 +414,154 @@ export default class PostTextBoxBase extends PureComponent {
                 }],
             );
         } else {
+            const currentMembersCount = this.props.currentChannelMembersCount;
+            const notificationsToChannel = this.props.enableConfirmNotificationsToChannel;
+            const toAllorChannel = this.textContainsAtAllAtChannel(value);
+
             if (value.indexOf('/') === 0) {
                 this.sendCommand(value);
-            } else {
-                const postFiles = files.filter((f) => !f.failed);
-                const post = {
-                    user_id: currentUserId,
-                    channel_id: channelId,
-                    root_id: rootId,
-                    parent_id: rootId,
-                    message: value,
-                };
-
-                actions.createPost(post, postFiles);
-
-                if (postFiles.length) {
-                    actions.handleClearFiles(channelId, rootId);
+            } else if (notificationsToChannel && currentMembersCount > NOTIFY_ALL_MEMBERS && toAllorChannel) {
+                if (this.props.isTimezoneEnabled) {
+                    const {data} = await this.props.actions.getChannelTimezones(this.props.currentChannel.id);
```

Comment:
definitely not keen on this one here.. it adds a delay sending the message and it will fetch it everytime, maybe this should be done during mount and when the channel changes

Reply:
Thanks for suggestion. I've made it so it only fetches on componenetdidmount and when channel changes. :)